src/js/components/AppListComponent.jsx

```diff
@@ -70,13 +70,19 @@ var AppListComponent = React.createClass({
     var sortKey = state.sortKey;
     var props = this.props;
 
-    return lazy(state.apps)
+    var appsSequence = lazy(state.apps);
+
+    if (props.filterText != null && props.filterText != "") {
+      appsSequence = appsSequence
+        .filter(function (app) {
+          return app.id.indexOf(props.filterText) !== -1;
```

Comment:
Here you can test the differences between resolving a sequence (with toArray or any other function that produces a real value) and not:
http://danieltao.com/lazy.js/#benchmark-results